corehq/apps/domain/tests/test_forms.py

```diff
@@ -215,3 +234,78 @@ def test_release_mode_settings_visible_and_saved_without_error(self):
         form.full_clean()
         saved = form.save(Mock(), self.domain_obj)
         self.assertEqual(True, saved)  # No error during form save
+
+
+class TestConfirmSubscriptionRenewalForm(TestCase, DomainSubscriptionMixin):
+    def setUp(self):
+        super().setUp()
+        self.domain = Domain(name='subscription-renewal', is_active=True)
+        self.domain.save()
+
+        username = 'clifford'
+        password = '*******'
+        self.user = WebUser.create(self.domain.name, username, password,
```

Comment:
we have some utils in `corehq.apps.accounting.tests.generator` for this, like `arbitrary_user`. I think that might help with some of the boiler plate setup here